src/basic/fd-util.h

```diff
@@ -126,3 +128,15 @@ static inline char *format_proc_fd_path(char buf[static PROC_FD_PATH_MAX], int f
 
 #define FORMAT_PROC_FD_PATH(fd) \
         format_proc_fd_path((char[PROC_FD_PATH_MAX]) {}, (fd))
+
+/* Determining a file descriptor's path via /proc is not reliable. Only use this function to provide better
+ * log messages when doing fd based path handling! */
+static inline const char *FORMAT_FD_PATH(int fd) {
+        static thread_local char buf[PATH_MAX] = {};
```

Comment:
4K on every single thread this might get loaded into is not OK

Reply:
also, if you use this twice in the same expression then things will fail badly. i.e.

```c
log_info("copying from %s to %s.", FORMAT_FD_PATH(sfd), FORMAT_FD_PATH(tfd));
```

This is just too likely to break

Reply:
Switched to a lazy heap allocated buffer to avoid the stack issue. I don't know how to fix the same expression issue though, but I figure that use case will be rare enough that it shouldn't matter.